src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -67,6 +69,9 @@ public class FilingsService {
   @Value("${FEATURE_FLAG_ENABLE_TRUSTS_CHIPS_1502023}")
   private boolean isTrustsSubmissionThroughWebEnabled;
 
+  @Value("${PUBLIC_API_IDENTITY_HASH_SALT:}")
```

Comment:
this looks like it will default to an empty string if the PUBLIC_API_IDENTITY_HASH_SALT property is not found, would it be better to remove the colon so that it will throw an error if the salt property is not found?

Reply:
> I hate to propose something completely different, but have you considered using Guava's Hashing utiltity? Guava is approved for use by CH and Hashing is very powerful. Most distinctively, it allows you to swap hash algorithms without substantially changing your code. https://github.com/google/guava/wiki/HashingExplained
> 
> You can find a use of Hashing in CHIPS code here: https://github.com/search?q=repo%3Acompanieshouse%2Fchips%20hashfunction&type=code

We need to use the same hash algo as the CHIPS to CHS to Public API. So SHA1 with all it's limitations. It's really only obfuscation of internal database Ids so not exactly security critical.

Reply:
> > I hate to propose something completely different, but have you considered using Guava's Hashing utiltity? Guava is approved for use by CH and Hashing is very powerful. Most distinctively, it allows you to swap hash algorithms without substantially changing your code. https://github.com/google/guava/wiki/HashingExplained
> > You can find a use of Hashing in CHIPS code here: https://github.com/search?q=repo%3Acompanieshouse%2Fchips%20hashfunction&type=code
> 
> We need to use the same hash algo as the CHIPS to CHS to Public API. So SHA1 with all it's limitations. It's really only obfuscation of internal database Ids so not exactly security critical.

I think you've attached your response to the wrong comment @AndreKramer7654 . @markpit 's question on the salt property still stands I think.